src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -967,6 +967,13 @@ internal void AddAndStartOutgoingReplication(ReplicationNode node, bool external
                 // this means that we were unable to retrieve the tcp connection info and will try it again later
                 return;
             }
+
+            if (Database == null)
+            {
+                // database was disabled
+                return;
+            }
+
             var outgoingReplication = new OutgoingReplicationHandler(this, Database, node, external, info);
```

Comment:
What if it happens inside `OutgoingReplicationHandler`?

Reply:
`ReplicationLoader` is already alive only in the context of a database, so I don't see why this is required.
If the DB was disposed one ms afterward?

Reply:
this code can run as not awaited [(code line),](https://github.com/ravendb/ravendb/blob/v4.2/src/Raven.Server/Documents/Replication/ReplicationLoader.cs#L725) so when we null the `Database` on dispose of `ReplicationLoader` this code could still run so we get NRE

Reply:
We need a better way to detect that.